proto/ibc/applications/interchain_accounts/controller/v1/tx.proto

```diff
@@ -26,3 +29,17 @@ message MsgRegisterAccount {
 message MsgRegisterAccountResponse {
   string channel_id = 1 [(gogoproto.moretags) = "yaml:\"channel_id\""];
 }
+
+// MsgSubmitTx defines the payload for MsgSubmitTx
+message MsgSubmitTx {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string   owner                   = 1;
+  string   connection_id           = 2 [(gogoproto.moretags) = "yaml:\"connection_id\""];
+  string   timeout                 = 3;
+  repeated google.protobuf.Any msg = 4;
```

Comment:
@damiannolan what's the reason for using `Any`s here? Shouldn't we just be taking in the ics27 [packet data](https://github.com/cosmos/ibc-go/blob/0fa2978b0edad338f995790b5ab3a517ff302351/modules/apps/27-interchain-accounts/types/packet.pb.go#L57)

Reply:
I think if an end user wants to use `x/group` or `x/gov` they can't really construct the packet data type as it expects a proto encoded `[]byte` of message data. If we use `Any` it can be defined in JSON and submitted for proposal execution. Essentially like it's done [here](https://github.com/cosmos/cosmos-sdk/blob/main/x/group/client/cli/tx.go#L559). We would have a proto-JSON encoded `MsgSubmitTx` which has a nested array of ICA messages to be executed on the host.

Reply:
Okay so this is an any which is decoded into packet data?

We cannot assume the counterparty uses any's for message delivery (this is SDK behaviour). I also suspect folks will want to make use of the memo field (it allows for a short term hack of enabling top level middleware for application handling, ie memo field: "swap:lp" would indicate, swap tokens then lp)

Reply:
I think it'd be problematic to use Any to represent the messages delivered to the counterparty chain. These messages would need to be registered on the controller chain since the SDK will verify the Any URL is registered on the interface registry. Does that make sense?

Reply:
> Okay so this is an any which is decoded into packet data?

I was thinking that this would be an array `Any`s which would be encoded into packet data using [`SerializeCosmosTx`](https://github.com/cosmos/ibc-go/blob/main/modules/apps/27-interchain-accounts/types/codec.go#L32). 

> These messages would need to be registered on the controller chain since the SDK will verify the Any URL is registered on the interface registry. Does that make sense?

This does make a lot of sense! It's a good point, do you know where in the SDK this would fail if for example you tried to provide an osmosis specific message type (swap) via a cosmos-hub(controller) group proposal?
I think this may be kind of problematic from a ux perspective...

Reply:
I believe the issue occurs because [`UnpackAny`](https://github.com/cosmos/cosmos-sdk/blob/main/codec/types/interface_registry.go#L217) will call [`UnpackInterfaces`](https://github.com/cosmos/cosmos-sdk/blob/main/codec/types/interface_registry.go#L294) which will try to [`unpackany`](https://github.com/cosmos/ibc-go/blob/main/modules/core/02-client/types/client.go#L42) the nested Any, but this will [fail to resolve](https://github.com/cosmos/cosmos-sdk/blob/main/codec/types/interface_registry.go#L243)